domain/model/state/controllerstate.go

```diff
@@ -1802,3 +1803,50 @@ AND    ot.type = $dbPermission.object_type
 	}
 	return nil
 }
+
+// GetAllActivatedModelsUUIDQuery returns a SQL statement that will get all the activated models uuids in the controller.
+func (st *State) GetAllActivatedModelsUUIDQuery() string {
+	return "SELECT uuid from model WHERE activated = true"
+}
+
+// GetActivatedModelUUIDs returns the subset of model uuids from the supplied list that are activated.
+// If no models associated with the uuids are activated then an empty slice is returned.
+func (st *State) GetActivatedModelUUIDs(ctx context.Context, uuids []coremodel.UUID) ([]coremodel.UUID, error) {
+	if len(uuids) == 0 {
+		return nil, nil
+	}
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	type modelUUID struct {
+		UUID coremodel.UUID `db:"uuid"`
+	}
+
+	stmt, err := st.Prepare(`
+SELECT uuid AS &modelUUID.uuid
+FROM   v_model
+WHERE uuid IN ($S[:])
+`, sqlair.S{}, modelUUID{})
+
+	if err != nil {
+		return nil, errors.Errorf("preparing select model activated status statement: %w", err)
+	}
+
+	args := sqlair.S(transform.Slice(uuids, func(s coremodel.UUID) any { return any(s) }))
+	modelUUIDs := []modelUUID{}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, args).GetAll(&modelUUIDs)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Capture(err)
```

Comment:
This error should be annotated with more information. It should be of the form "verb what were we doing: %w".

Reply:
Refactored with thanks

Reply:
Refactored.